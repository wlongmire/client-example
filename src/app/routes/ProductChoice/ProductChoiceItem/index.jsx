import React, { PropTypes } from 'react'
import { connect } from 'react-redux'

import { push } from 'react-router-redux'

import {
  CHANGE_SUBMISSION,
  SUBMISSION_STATUS
} from 'app/constants/submission'

import mx from 'app/utils/MixpanelInterface'

export function ProductChoiceItem(props) {
  const {
    type,
    name,
    description,
    broker
  } = props

  return (
    <div
      data-type={type}
      className="selectionCard"
      onClick={() => {
        props.dispatch({
          type: CHANGE_SUBMISSION,
          payload: {
            submission: { type, status: SUBMISSION_STATUS.CLEARANCE }
          }
        })

        mx.customEvent(
          'submission',
          'create',
          {
            Type: type
          }
        )

        props.dispatch(push('/clearance'))
        /* props.dispatch(push("/form")) */
      }}
    >
      <h1>{name}</h1>
      <p>{description}</p>
    </div>)
}
ProductChoiceItem.propTypes = {
  dispatch: PropTypes.func,
  type: PropTypes.string,
  name: PropTypes.string,
  description: PropTypes.string,
  broker: PropTypes.string
}


export default connect()(ProductChoiceItem)