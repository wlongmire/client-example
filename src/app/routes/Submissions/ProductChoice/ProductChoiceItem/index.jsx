import React from 'react'
import PropTypes from 'prop-types'

import { connect } from 'react-redux'

import { push } from 'react-router-redux'

import { Row, Col, Button } from 'react-bootstrap'

import {isNullOrEmpty} from '../../../../utils/utilities'

import {
  CHANGE_SUBMISSION,
  SUBMISSION_STATUS
} from 'app/constants/submission'

import mx from 'app/utils/MixpanelInterface'

export function ProductChoiceItem(props) {
  const {
    type,
    name,
    description,
    broker,
    subtitle
  } = props

  return (
    <div data-type={type}>
        <div className="productContainer">
          <h4>{name}</h4>
          <span>{subtitle}&nbsp;</span>
          <Button onClick={() => {
            props.dispatch({
              type: CHANGE_SUBMISSION,
              payload: {
                submission: { type, status: SUBMISSION_STATUS.CLEARANCE }
              }
            })

            mx.customEvent(
              'submission',
              'create',
              {
                Type: type
              }
            )

            props.dispatch(push('/clearance'))
            /* props.dispatch(push("/form")) */
          }}
          >Start Submission</Button>
        </div>
    </div>)
}
ProductChoiceItem.propTypes = {
  dispatch: PropTypes.func,
  type: PropTypes.string,
  name: PropTypes.string,
  description: PropTypes.string,
  broker: PropTypes.string,
  subtitle: PropTypes.string
}


export default connect()(ProductChoiceItem)