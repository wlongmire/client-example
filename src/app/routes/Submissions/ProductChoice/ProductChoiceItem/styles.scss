.productContainer {
  h4 {
    color:#CFD2CD
  }
  .btn{

  }
};