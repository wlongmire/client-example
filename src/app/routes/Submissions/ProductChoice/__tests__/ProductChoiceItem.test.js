/* eslint-env node, jest */

import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import { ProductChoiceItem } from '../ProductChoiceItem'

describe('>>> ProductChoiceItem Component --- Snapshot', () => {
  it('+++ OCP in product choice page', () => {
    const renderedValue = shallow(
      <ProductChoiceItem
        broker={'test-2222-7fd-b3ff-4fd3-9fc2-e752b9f5b002'}
        description={'Random Description text'}
        name={'Owners and Contractors Protective'}
        type={'ocp'}
      />)
    expect(toJson(renderedValue)).toMatchSnapshot()
  })
})

describe('>>> ProductChoiceItem Component --- Snapshot', () => {
  it('+++ OI in product choice page', () => {
    const renderedValue = shallow(
      <ProductChoiceItem
        broker={'test-2222-7fd-b3ff-4fd3-9fc2-e752b9f5b002'}
        description={'Random Description text'}
        name={'Owners Interest'}
        type={'oi'}
      />)
    expect(toJson(renderedValue)).toMatchSnapshot()
  })
})

// describe('>>> Quote Component with Email Pending --- Snapshot', () => {
//   it('+++capturing Snapshot with', () => {
//     const renderedValue = shallow(
//       <Knockout
//         submission={submission}
//         emailStatus={'PENDING'}
//         ratings={bundleRatings}
//       />)
//     expect(toJson(renderedValue)).toMatchSnapshot()
//   })
// })