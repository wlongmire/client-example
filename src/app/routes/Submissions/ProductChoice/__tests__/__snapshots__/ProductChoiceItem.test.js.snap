// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`>>> ProductChoiceItem Component --- Snapshot +++ OCP in product choice page 1`] = `
<div
  className="selectionCard"
  data-type="ocp"
  onClick={[Function]}
>
  <h1>
    Owners and Contractors Protective
  </h1>
  <p>
    Random Description text
  </p>
</div>
`;

exports[`>>> ProductChoiceItem Component --- Snapshot +++ OI in product choice page 1`] = `
<div
  className="selectionCard"
  data-type="oi"
  onClick={[Function]}
>
  <h1>
    Owners Interest
  </h1>
  <p>
    Random Description text
  </p>
</div>
`;
